File: TicTacToe/TicTacToe/GameMain.vb
Module GameMain

    Sub Main()
        MainLoop()

    End Sub

    Sub MainLoop()
        Console.WriteLine("/-/ ■Tic Tac Toe Game! /-/")
        Console.WriteLine()

        Dim mainPlayer As New Player(UserType.User)
        Dim cpuPlayer As New Player(UserType.CPU)

        Dim board As New GameBoard()

        Dim judge As New Judge(board)

        board.GameStart()
        While (board.IsFinish() <> True)

            mainPlayer.Play(board)
            board.Display()

            If (judge.JudgeGame(mainPlayer)) Then
                Exit While
            End If

            cpuPlayer.Play(board)
            board.Display()

            If (judge.JudgeGame(cpuPlayer)) Then
                Exit While
            End If

        End While
        board.GameFinish()

    End Sub

End Module

File: TicTacToe/TicTacToe/GameMain.cs
using System;
using System.Threading;

namespace TicTacToe
{
    class GameMain
    {

        public GameMain()
        {
        }

        static void Main()
        {
            new GameMain().MainLoop();
        }

        void MainLoop()
        {
            Console.WriteLine("### Tic Tac Toe game! :-) ###");
            Console.WriteLine();

            Player mainPlayer = new Player(PlayerType.USER);
            Player cpuPlayer = new Player(PlayerType.CPU);

            GameBoard board = new GameBoard();
            Judge judge = new Judge(board);

            board.startgame();
            while (board.IsFinished == false) 
            {
                board.Display();

                mainPlayer.Play(board);
                if (judge.judgegame(mainPlayer) == true) {
                    break;
                }

                cpuPlayer.Play(board);
                if (judge.judgegame(cpuPlayer) == true)
                {
                    break;
                }
            }
            board.endgame();
            Thread.Sleep(3000);   
        }

    }
}
